test/ERC721AStartOne.test.js

```diff
@@ -0,0 +1,341 @@
+const { expect } = require('chai');
+const { constants } = require('@openzeppelin/test-helpers');
+const { ZERO_ADDRESS } = constants;
+
+const RECEIVER_MAGIC_VALUE = '0x150b7a02';
+const GAS_MAGIC_VALUE = 20000;
+
+describe('ERC721AStartOne', function () {
+  beforeEach(async function () {
+    this.ERC721A = await ethers.getContractFactory('ERC721AStartOneMock');
```

Comment:
I think a better approach to our tests would be parameterize this class variable to change between the 0-index and 1-index contracts so we don't have to duplicate most of the code.

You would then have specific tests for `ownerOf` and tokenId. You would probably need a separate context for 1-index specific test cases.

I want to minimize the work we need to do anytime we add/change some functionality in our contract.

Reply:
Agree, to remove most of the duplicate test code. This is a good hint, I will look into that. When I copied them I didn't know better. Will try to adapt it today.